src/main/java/updater/UpdateManager.java

```diff
@@ -134,29 +134,30 @@ private boolean initUpdate() {
         updateConfig.setLastAppUpdateDownloadSuccessful(false);
         saveUpdateConfig();
 
-        return extractJarUpdater();
+        try {
+            extractJarUpdater();
+        } catch (IOException e) {
+            logger.error("Failed to extract Jar Updater", e);
+            return false;
+        }
+
+        return true;
     }
 
-    private boolean extractJarUpdater() {
+    private void extractJarUpdater() throws IOException {
         logger.info("Extracting jarUpdater");
         File jarUpdaterFile = new File(JAR_UPDATER_APP_PATH);
 
-        try {
-            jarUpdaterFile.createNewFile();
-        } catch (IOException e) {
-            logger.error("Can't create empty file for jarUpdater");
-            return false;
+        if (!jarUpdaterFile.exists()) {
+            assert jarUpdaterFile.createNewFile();
```

Comment:
I'm not sure assert is the right thing to do here. File system problems are not programmer problems.